src/solvers/smt2_incremental/convert_expr_to_smt.cpp

```diff
@@ -325,8 +325,12 @@ convert_expr_to_smt(const binary_relation_exprt &binary_relation)
 
 static smt_termt convert_expr_to_smt(const plus_exprt &plus)
 {
-  UNIMPLEMENTED_FEATURE(
-    "Generation of SMT formula for plus expression: " + plus.pretty());
+  for(const auto &op : plus.operands())
+  {
+    PRECONDITION(can_cast_type<bitvector_typet>(op.type()));
```

Comment:
My rationale for this is that for now we do not support them (or more precisely, the operation is not supported for any type other than bitvectors) so I'm guarding for that type in the precondition. 

That check would be relaxed when needed (when support for the conversion of integers, rationals, etc are added).

Do you mean that you want a special message to that effect?

Reply:
I agree that this should have appropriate `INVARIANT`s. My argument is that this should be done based on an `UNIMPLMENTED` condition at the end of this function, rather than a `PRECONDITION` at the start. So the parts we have implemented are reached when conditions implemented are satisfied and we fall through to the `UNIMPLEMENTED` macro, if the given expression is not supported by any of the cases we have implemented.

My comment relates to how easy it will be to extend the implementation you have added, not really the ease of understanding it. Therefore I don't think additional messages would be helpful here.

Reply:
Looks good to me.